php/libraries/Database.class.inc

```diff
@@ -978,7 +978,9 @@ class Database
         }
 
         foreach ($dataArray as $key => $item ) {
-            if ($item===null || $item==='') {
+            if ($item===null) {
```

Comment:
@driusan correct me if i'm wrong but isn't this function also used to build the $set array for UPDATE statements ?

in that case it would generate `UPDATE table_name SET variable1=5, variable 2 IS NULL, variable3...`

Reply:
@ridz1208  @driusan @xlecours  You are right about that. It should not be the case now as it was just not permitted before. 
Does adding an optional parameter for NULL value, `link = null` with possible value 'null' default, 'IS' and 'IS NOT'.
This should be an array of the same size of $param as it is possible that some parameter need to be IS NULL while other migth be IS NOT NULL.

This way we will

Reply:
@PapillonMcGillI think  @ridz1208 is talking about the `SET` part an `UPDATE` query, not the `WHERE` part. The `SET` part never needs "IS NULL/IS NOT NULL" while the `WHERE` part always does.. shouldn't a single extra bool parameter be sufficient?

Reply:
@driusan Yes, I could add a parameter  $clause = 'set' by default and allow 'where', but if in a where clause we later need IS NOT NULL instead of IS NULL, we will still be stuck. So with an array (NULL by default), we could add for each parameter where we need a comparison with null to be either IS NULL or IS NOT NULL as required and for those parameter where it is not a comparison with null or it is in a SET clause, leave that parameter as null and '=' will be used.

Reply:
I think we should have an `_implodeSetWithKeys()` and an `_implodeWhereWithKeys()`

Reply:
There's already 2 calls (one for set, and one for where), so you still don't need an array (but @ridz1208's proposal of 2 private functions with different names still makes more sense than an extra param to me..)... but now that I'm looking at the code, the update path is only using implodeWithKeys for printing the query. It uses _implodeAsPrepared for the actual query that it runs.

(Delete should also be using the prepared variant, I'm not sure why it isn't.)

Reply:
@driusan using _implodeAsPrepared() will be hard for "IS NULL" or "IS NOT NULL" as PDO do not support that in prepared statements. 
So I will have to stick with _implodeWithKeys(). But adding a variable $clause = 'set' in the function definition will not solve the future possibility of "IS NOT NULL".  Unless we use a dirty trick like using PHP_EOL constant to convert to IS NOT NULL but that might induce other problem.